### PR TITLE
[Testing]: User Endpoints

### LINKED ISSUE #39: [Testing]: User Endpoints
Implementation of unit tests for the user endpoints in the backend to ensure the stability and reliability of our backend services.

### PR DESCRIPTION
**Pull request by xvxnoah**
To test these endpoints, we will focus on the following testing parameters:

1. Happy Path: Testing when everything goes as expected.

2. Boundary/Edge Cases: Testing with the extremes.

3. Error Cases: Testing where we expect errors or failures.

### CODE REVIEW

**Comment by xvxnoah** [comment]
To test correctly, we have to add 2 new environment variables called: 
- DB_TEST="userttests"
- TEST_MODE=True

To run the tests, we just have to run the script run_tests.py.

File: app/routers/user_router.py [hunk 1]
```diff
@@ -4,16 +4,28 @@
 from app.utils.token import create_access_token
 from bson import ObjectId
 
+# Suppress UserWarning from pydantic
+warnings.filterwarnings("ignore", category=UserWarning, module="pydantic")
```
Comment (BlessedLongsword) [inline app/routers/user_router.py:8]: What is the purpose of this, asking for knowledge more than anything.

**Review by BlessedLongsword — DISMISSED** [review]

**Comment by BlessedLongsword** [comment]
I guess I will leave the conflict resolution to @andreu-vall, since the conflicts are related to his code. By the way @andreu-vall I am not sure about the pyjwt requirement in requirement.in, I don't really need it for my system.

File: app/routers/user_router.py (hunk 1 above)
Comment (xvxnoah) [reply app/routers/user_router.py:8]: When creating a user, related to the way we handle the ids, there was always a warning related to pydantic.

**Review by BlessedLongsword — DISMISSED** [review]
After the last commit everything is working properly. Very good job! Now we have to wait for @andreu-vall to resolve the conflicts.

**Comment by andreu-vall** [comment]
I thought we already talked about it, but bcrypt and pyjwt are needed for encryption. Maybe somehow your systems imports it somehow from somewhere else, but they are needed. Have you checked from where your modules are coming from? If I deploy it in Linux I could try to delete the requirements, but for now I need them and I don't understand why u hate them

**Comment by BlessedLongsword** [comment]
Okay, sorry for being so oppressive about it haha. Anyway, if you could take a look and resolve the conflicts it'd be great. Let me know if you can't and I will do it.

**Comment by andreu-vall** [comment]
I have already solved it from GitHub, but I'm cloning it again to check that it still works

**Review by BlessedLongsword — APPROVED** [review]
Seems good finally haha. Can merge
